### ISSUE TITLE
Classes which are provided in the "Excludes" should be removed from the coverage percentage

### ISSUE THREAD

**Issue by clementfung** [jacoco/eclemma#84]
In Eclipse:
Window -> Preferences -> Java -> Code Coverage

Specifying a class within the "Coverage Runtime: Excludes" section causes the coverage for that class to become 0%. However, the "Missed Instructions" for those classes are still included in the coverage report.

In my opinion, this option should also remove these lines from the denominator.

Thank you!

**Comment by deepy** [comment]
Has there been any updates in regards to this?
I spent a while trying to figure out why "Coverage Runtime: Excludes" wasn't working before I found this.

**Comment by jandroalvarez** [comment]
Any news about this issue? This is very annoying while trying to find out you percentage of coverage. From my point of view, "Excludes" is useless if you don't get a percentage that doesn't take into account the excluded classes.

**Comment by davidcok** [comment]
I vote for this as well.  But in addition, the exclusions should be specified in the Coverage Configuration so that they are shared across a work group and one can have different Configurations with different exclusions.  Excluding classes as a workspace property makes little sense to me.

**Comment by jbravo-tejeda** [comment]
Hello guys! Is this happening? Is commenting here a waste of time? It's been over two years since the first post here and nobody from the dev team has done any activity on this request other than adding a label? Even if it isn't happening, or if it's some time down the line, can we get a comment from someone acknowledging the request? Thanks and have a great day!

**Comment by Godin** [comment]
@jbravo-tejeda we are currently in the process of transferring project to Eclipse foundation - more about this at https://groups.google.com/d/msg/jacoco/fMpiBktkKZM/9urtDcJgCAAJ Once this will be done - we'll review backlog of open tickets.

More about dev activity and priorities see https://github.com/jacoco/jacoco/issues/15#issuecomment-241092025 , which was for JaCoCo, but applicable here as well. So yes - such comments can be seen as a kind of waste of time (both yours and ours): instead of asking "when" and waiting years, far better to come with a contribution of a patch.

Thank you for your understanding.

**Comment by jbravo-tejeda** [comment]
@Godin Thank you! I'm sure the comment you just posted will be appreciated by the community, at the very least for this request.

**Comment by monism** [comment]
Hi All,

I am trying to exclude a class that i know have coverage in it. Once i put that to the excludes preference, on a new coverage run they'll still be there in the coverage window, but they'll come up as 0% and will all be in red.

Is there a solution which to solve this , i need to exclude a class and not be showing in report at all

**Comment by marchof** [comment]
@monism No, EclEmma does not support this, that's why this feature request is here. Cross-posting in various issues will also not help to get this feature implement. Please stop it and use our mailing list for questions.

**Comment by egry35** [comment]
+1

**Comment by david0603** [comment]
+1

**Comment by albertrobi** [comment]
+1

**Comment by darkoromanov** [comment]
+1

**Comment by jonathanqbo** [comment]
+1
